src/js/cf-expandables.js

```diff
@@ -5,110 +5,185 @@
-        this.toggle();
-        if ( accordion ) {
-          $siblings.each( function( index, sibling ) {
-            sibling.collapse();
-          });
-        }
-      };
+    initUI: function initUI() {
+      var $el = this.$el;
 
-      this.toggle = function() {
-        if ( $target.attr('aria-pressed') === 'true' ) {
-          this.collapse();
-        } else {
-          this.expand();
-        }
-      };
-
-      this.expand = function( duration ) {
-        $cueOpen.css( 'display', 'none' );
-        $cueClose.css( 'display', 'inline' );
-        $content.attr( 'aria-expanded', 'true' );
-        $target.attr( 'aria-pressed', 'true' );
-        if ( typeof duration === 'undefined' ) {
-          duration = $.fn.expandable.calculateExpandDuration( $content.height() );
+      for ( var uiKey in this.ui ) {
+        if ( this.ui.hasOwnProperty( uiKey ) ) {
+          this['$' + uiKey] = $el.find( this.ui[uiKey] + ':first' );
         }
-        $this.addClass('expandable__expanded');
-        $content.slideDown({
-          duration: duration,
-          easing: 'easeOutExpo'
-        });
-      };
-
-      this.collapse = function( duration ) {
-        $cueOpen.css( 'display', 'inline' );
-        $cueClose.css( 'display', 'none' );
-        $content.attr( 'aria-expanded', 'false' );
-        $target.attr( 'aria-pressed', 'false' );
-        if ( typeof duration === 'undefined' ) {
-          duration = $.fn.expandable.calculateCollapseDuration( $content.height() );
-        }
-        $this.removeClass('expandable__expanded');
-        $content.slideUp({
-          duration: duration,
-          easing: 'easeOutExpo'
-        });
-      };
+      }
 
-      this.init();
+      this.$target.attr( 'aria-controls', this.$content.attr( 'id' ) );
 
-    });
+      if ( $el.hasClass( this.expandedClass ) ) {
+        this.expand( 0 );
+      } else {
+        this.collapse( 0 );
+      }
+    },
 
-  };
+    initEvents: function initEvents() {
+      var toggle = $.proxy( _throttle, this, 450, this.toggle );
+      this.$el.on( 'click', '.expandable_target', toggle );
+    },
 
-  $.fn.expandable.calculateExpandDuration = function( height ) {
-    return $.fn.expandable.constrainValue( 450, 900, height * 4 );
-  };
+    toggle: function toggle( event ) {
+      event.preventDefault();
+      event.stopPropagation();
 
-  $.fn.expandable.calculateCollapseDuration = function( height ) {
-    return $.fn.expandable.constrainValue( 350, 900, height * 2 );
-  };
+      if ( this.isInAccordion ) {
+        this.$el.siblings( '.expandable' ).each( function( index, sibling ) {
+          sibling.collapse();
+        } );
+      }
+
+      if ( this.isExpanded ) {
+        this.collapse();
+      } else {
+        this.expand();
+      }
+    },
+
+    expand: function expand( duration ) {
+      var $content = this.$content;
+      this.$cueOpen.css( 'display', 'none' );
+      this.$cueClose.css( 'display', 'inline' );
```

Comment:
Can these values use a class instead, like `.expandable_cue-open` and `.expandable_cue-close` and managed in helper functions that work on the cue element? This probably requires a change in the underlying structure so there's just a `$cue` instead of an `$cueOpen` and `$cueClose`. E.g.:

```
function _openCue( target ) {
    target.addClass( '.expandable_cue-open' );
    target.removeClass(  '.expandable_cue-close' );
}

function _closeCue( target ) {
    target.removeClass( '.expandable_cue-open' );
    target.addClass(  '.expandable_cue-close' );
}
```

And then L:111 and L:112 would look like `_openCue( this.$cue )`, and L:132 and L:133 would look like `_closeCue( this.$cue )`.